python/triton/language/extra/cuda/_experimental_tma.py

```diff
@@ -0,0 +1,116 @@
+from triton.language import core
+from triton.language import semantic
+from triton._C.libtriton import ir
+
+from typing import Sequence
+
+__all__ = [
+    "experimental_device_tensormap_create1d",
+    "experimental_device_tensormap_create2d",
+    "experimental_tensormap_fenceproxy_acquire",
+]
+
+
+def _set_tma_dtype(desc_ptr: core.tensor, element_ty: core.dtype, _builder: ir.builder):
+    if element_ty.primitive_bitwidth == 8:
+        semantic.tensormap_replace_elemtype(desc_ptr, 0, _builder)
+    elif element_ty.primitive_bitwidth == 16:
+        semantic.tensormap_replace_elemtype(desc_ptr, 1, _builder)
+    elif element_ty.primitive_bitwidth == 32:
+        semantic.tensormap_replace_elemtype(desc_ptr, 2, _builder)
+    else:
+        raise ValueError("element_ty must be a primitive of size 1, 2, or 4 bytes but got")
+
+
+@core.builtin
+def experimental_device_tensormap_create1d(
+    desc_out: core.tensor,
+    template_desc: core.tensor,
+    global_address: core.tensor,
+    load_size: core.tensor,
+    global_size: core.tensor,
+    element_ty: core.dtype,
+    _builder: ir.builder,
+):
+    load_size = core._constexpr_to_value(load_size)
+    global_size = core._to_tensor(global_size, _builder)
+    element_ty = core._constexpr_to_value(element_ty)
+
+    local_desc = semantic.tensormap_allocate(template_desc, _builder)
+    semantic.tensormap_replace_global_address(local_desc, global_address, _builder)
+    _set_tma_dtype(local_desc, element_ty, _builder)
+    semantic.tensormap_replace_rank(local_desc, 0, _builder)
+    semantic.tensormap_replace_box_dim(local_desc, 0, load_size, _builder)
+    semantic.tensormap_replace_global_dim(local_desc, 0, global_size, _builder)
+    semantic.tensormap_replace_element_stride(local_desc, 0, 1, _builder)
+    semantic.tensormap_replace_interleave_layout(local_desc, 0, _builder)
+    semantic.tensormap_replace_swizzle_mode(local_desc, 0, _builder)
+    semantic.tensormap_replace_fill_mode(local_desc, 0, _builder)
+
+    semantic.tensormap_cp_fenceproxy(desc_out, local_desc, builder=_builder)
+    semantic.tensormap_deallocate(local_desc, _builder)
```

Comment:
I think this kind of lowering would fit better in TTGIR, this would avoid exposing as much details to TTIR and would allow better re-use of the existing TTGIR operations instead of having to add so many new ops to TTIR.

We could just have a TTIR op that maps `experimental_device_tensormap_create` kind of signature then lower it in TTGIR to a shared memory alloc with some local_stores